arroyo-api/src/metrics.rs

```diff
@@ -41,41 +44,41 @@ static METRICS_CLIENT: Lazy<Client> = Lazy::new(|| {
     prometheus_http_query::Client::from(client, &prometheus_endpoint).unwrap()
 });
 
-pub(crate) async fn get_metrics(
-    job_pub_id: String,
-    auth: AuthData,
-    client: &impl GenericClient,
-) -> Result<JobMetricsResp, Status> {
-    let job_id = api_queries::get_pipeline_job()
-        .bind(client, &auth.organization_id, &job_pub_id)
-        .one()
-        .await
-        .map_err(log_and_map)?
-        .id;
-
-    // validate that the job exists and user can access it
-    let job_details = jobs::get_job_details(&job_id, &auth, client).await?;
-
+/// Get a job's metrics
+#[utoipa::path(
+    get,
+    path = "/v1/pipelines/{pipeline_id}/jobs/{job_id}/operator_metric_groups",
```

Comment:
Sure, I guess, but to follow the REST convention of plural nouns implying being collections then we'll need to name this struct/object like this:
```
pub struct OperatorMetric {
    pub operator_id: String,
    pub metric_groups: Vec<MetricGroup>,
}
```
...which is weird because it's not a single _metric_. This kind of naming problem is kind of a consequence of too many levels of nesting in api objects.

Reply:
I'll try to come up with something better though.

Reply:
I actually think this is the best option even though it's long. Let me know if you have any other ideas.

Reply:
sounds good